code/renderers/vue3/src/decorateStory.ts

```diff
@@ -49,10 +51,13 @@ export function decorateStory(
       let story: VueRenderer['storyResult'] | undefined;
 
       const decoratedStory: VueRenderer['storyResult'] = decorator((update) => {
+        // we should update the context with the update object from the decorator in reactive way
+        // so that the story will be re-rendered with the new context
         story = decorated({
           ...context,
           ...sanitizeStoryContextUpdate(update),
         });
+        if (update) updateReactiveContext(context, update);
         return story;
```

Comment:
Shouldn't we pass the updated context into `story`?

Reply:
it is passed already i guess but i need to update the reactive context ( used to track the state for vue ).
@tmeasday  i was little bit fighting to get how storybook rendering system works by myself, i guess i got 90% but sure you will guide me to  get 10% remaining .

Reply:
Have you seen the Svelte implementation? @chakAs3 
We were thinking that the proper way for Vue maybe similar.
We might need to treat every decorator as an actual vue component.

cc @JReinhold